static/routing-tree.js

```diff
@@ -166,6 +193,48 @@ function massage(root, receivers) {
     }
   }
 
+  // PromQL matcher syntax check
+  if (root.matchers) {
+    root.matchers.forEach((matcher) => {
+      let o = {};
+      let matchedExpressions;
+
+      while ((matchedExpressions = promQlRegExp.exec(matcher)) !== null) {
+        let [match, label, selector, value] = matchedExpressions;
+        o.name = label;
+
+        switch (selector) {
+          case "=~":
+            if (/[*+?\\.^\[\]$&|]+/.test(value)) {
+              o.value = new RegExp("^(?:" + value + ")$");
+              o.op = Matcher.RE;
+            } else {
+              // If no special characters in RegexMatcher than we can consider it as Equal Matcher
```

Comment:
Is there a reason why we wouldn't just still treat it as a regex if the selector is `=~`? If it ends up matching just like an equality matcher, that's fine...

Reply:
This is probably done in terms of saving resources, but it looks strange and confusing, and there is almost no benefit in terms of saving. You're right